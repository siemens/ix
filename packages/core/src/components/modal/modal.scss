/*
 * SPDX-FileCopyrightText: 2023 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

@import '~bootstrap/scss/_functions';
@import '~bootstrap/scss/mixins/_breakpoints.scss';
@import '~bootstrap/scss/_variables.scss';

@import 'common-variables';

:host {
  position: relative;
  width: 100vw;
  height: 100vh;

  .modal {
    display: flex;

    &.animation {
      animation: modal-fade $medium-time ease-in-out;
    }

    &.backdrop {
      background-color: var(--theme-color-lightbox);
      backdrop-filter: blur(4px);
      transition: none;

      &.animation {
        transition: backdrop-filter $short-time ease-in-out;
      }
    }

    .modal-dialog {
      flex-grow: 1;
    }

    &.scrollable {
      .modal-content {
        max-height: 100%;
        .modal-body {
          overflow-y: auto;
        }
      }
    }

    .modal-content {
      box-shadow: var(--theme-modal--box-shadow);
      flex-direction: row;
      background-color: var(--theme-modal--background);
      border: var(--modal--border-thickness) solid
        var(--theme-modal--border-color);
      padding: $large-space;
      max-height: 100vh;
      border-radius: var(--theme-default-border-radius);

      .state-icon-container {
        margin-inline-end: $default-space;
      }

      .slot-container {
        flex-grow: 1;
      }
    }

    @include media-breakpoint-up(sm) {
      .modal-dialog:not(.modal-dialog-centered) {
        margin: 2.5rem auto;
      }
    }

    @include media-breakpoint-down(sm) {
      .modal-dialog {
        height: 100%;
        margin: 0;

        .modal-content {
          height: 100%;
          border-radius: 0;
        }
      }
    }

    @keyframes modal-fade {
      from {
        opacity: 0%;
      }
      to {
        opacity: 100%;
      }
    }

    @keyframes slide-down {
      from {
        top: -100%;
      }
      to {
        top: 32px;
      }
    }
  }
}
