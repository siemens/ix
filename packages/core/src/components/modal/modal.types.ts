/*
 * SPDX-FileCopyrightText: 2025 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
export type IxModalFixedSize = '360' | '480' | '600' | '720' | '840';
export type IxModalDynamicSize = 'full-width' | 'full-screen';
export type IxModalSize = IxModalFixedSize | IxModalDynamicSize;
