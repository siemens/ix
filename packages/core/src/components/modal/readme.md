<!--
SPDX-FileCopyrightText: 2022 Siemens AG

SPDX-License-Identifier: MIT
-->

# ix-modal

<!-- Auto Generated Below -->


## Properties

| Property           | Attribute            | Description                                                                                                                                                                  | Type                                                                                      | Default            |
| ------------------ | -------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------- | ------------------ |
| `animation`        | `animation`          | Should the modal be animtated                                                                                                                                                | `boolean`                                                                                 | `true`             |
| `ariaDescribedBy`  | `aria-described-by`  |                                                                                                                                                                              | `string`                                                                                  | `undefined`        |
| `ariaLabelledBy`   | `aria-labelled-by`   |                                                                                                                                                                              | `string`                                                                                  | `'modal-title'`    |
| `backdrop`         | `backdrop`           | Adds a dimming layer to the modal. This should only be used when it it necessary to focus the user's attention to the dialog content (e.g. errors, warnings, complex tasks). | `"static" \| boolean`                                                                     | `true`             |
| `backdropClass`    | `backdrop-class`     | Backdrop class                                                                                                                                                               | `string`                                                                                  | `undefined`        |
| `beforeDismiss`    | --                   | BeforeDismiss callback                                                                                                                                                       | `(reason?: any) => boolean \| Promise<boolean>`                                           | `undefined`        |
| `centered`         | `centered`           | Centered modal                                                                                                                                                               | `boolean`                                                                                 | `false`            |
| `content`          | `content`            | Content of modal                                                                                                                                                             | `HTMLElement \| string`                                                                   | `undefined`        |
| `headerTitle`      | `header-title`       | Header title                                                                                                                                                                 | `string`                                                                                  | `undefined`        |
| `icon`             | `icon`               | Optional icon displayed next to the title                                                                                                                                    | `string`                                                                                  | `undefined`        |
| `iconColor`        | `icon-color`         | Color of the header {@see ix-icon}                                                                                                                                           | `"color-alarm" \| "color-info" \| "color-std-text" \| "color-success" \| "color-warning"` | `'color-std-text'` |
| `keyboard`         | `keyboard`           | ESC close modal dialog                                                                                                                                                       | `boolean`                                                                                 | `true`             |
| `modalDialogClass` | `modal-dialog-class` | Modal dialog class                                                                                                                                                           | `string`                                                                                  | `undefined`        |
| `scrollable`       | `scrollable`         | Modal scollable                                                                                                                                                              | `boolean`                                                                                 | `true`             |
| `size`             | `size`               | Modal size                                                                                                                                                                   | `"lg" \| "sm" \| "xl"`                                                                    | `'sm'`             |
| `windowClass`      | `window-class`       | Window class                                                                                                                                                                 | `string`                                                                                  | `undefined`        |


## Events

| Event       | Description     | Type               |
| ----------- | --------------- | ------------------ |
| `closed`    | Modal closed    | `CustomEvent<any>` |
| `dismissed` | Modal dismissed | `CustomEvent<any>` |


## Methods

### `close(result: any) => Promise<void>`

Close modal

#### Returns

Type: `Promise<void>`



### `dismiss(reason?: any) => Promise<void>`

Dismiss modal instance

#### Returns

Type: `Promise<void>`




----------------------------------------------


