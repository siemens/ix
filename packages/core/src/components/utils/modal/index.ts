/*
 * SPDX-FileCopyrightText: 2023 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export { showModalLoading } from './loading';
export { showMessage } from './message';
export {
  closeModal,
  dismissModal,
  ModalConfig,
  ModalInstance,
  showModal,
} from './modal';
