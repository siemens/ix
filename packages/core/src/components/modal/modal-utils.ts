/*
 * SPDX-FileCopyrightText: 2023 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { NotificationColor } from '../utils/notification-color';
import { TypedEvent } from '../utils/typed-event';

export interface ModalConfig<TReason = any> {
  animation?: boolean;
  ariaDescribedBy?: string;
  ariaLabelledBy?: string;
  backdrop?: boolean | 'static';
  backdropClass?: string;
  beforeDismiss?: (reason?: TReason) => boolean | Promise<boolean>;
  centered?: boolean;
  container?: string | HTMLElement;
  content: string | HTMLElement;
  keyboard?: boolean;
  modalDialogClass?: string;
  scrollable?: boolean;
  size?: 'sm' | 'lg' | 'xl';
  title: string;
  windowClass?: string;
  icon?: string;
  iconColor?: NotificationColor;
}

export interface ModalInstance<TReason = any> {
  htmlElement: HTMLIxModalElement;
  onClose: TypedEvent<TReason>;
  onDismiss: TypedEvent<TReason>;
}

function getModalContainer() {
  const containerList = Array.from(
    document.querySelectorAll('ix-modal-container')
  );
  const [container] = containerList;
  if (containerList.length > 1) {
    console.warn(
      'Multiple modal containers were found. The one instatiated first will be used.'
    );
    return container;
  }
  if (!container) {
    const modalContainer = document.createElement('ix-modal-container');
    document.body.appendChild(modalContainer);

    return modalContainer;
  }
  return container;
}

export async function modal<T = any>(
  config: ModalConfig<T>
): Promise<ModalInstance<T>> {
  const modalContainer = getModalContainer();
  const modalInstance = await modalContainer.showModal<T>(config);

  return modalInstance;
}

function getIxModal(element: Element) {
  return element.closest('ix-modal');
}

export function closeModal<TClose = any>(
  element: Element,
  closeResult: TClose
) {
  getIxModal(element).close(closeResult);
}

export function dismissModal(element: Element, dismissResult?: any) {
  getIxModal(element).dismiss(dismissResult);
}
