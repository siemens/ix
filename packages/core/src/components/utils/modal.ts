/*
 * SPDX-FileCopyrightText: 2023 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { MessageContent } from '../modal-message/modal-message';
import { IxModalSize } from '../modal/dialog';
import { getCoreDelegate, resolveDelegate } from './delegate';
import { NotificationColor } from './notification-color';
import { TypedEvent } from './typed-event';

export interface ModalConfig<TReason = any, C = any> {
  animation?: boolean;
  ariaDescribedBy?: string;
  ariaLabelledBy?: string;
  backdrop?: boolean;
  closeOnBackdropClick?: boolean;
  beforeDismiss?: (reason?: TReason) => boolean | Promise<boolean>;
  centered?: boolean;
  container?: string | HTMLElement;
  content: C;
  keyboard?: boolean;
  modalDialogClass?: string;
  scrollable?: boolean;
  size?: IxModalSize;
  title?: string;
  windowClass?: string;
  icon?: string;
  iconColor?: NotificationColor;
}

export interface ModalInstance<TReason = any> {
  htmlElement: HTMLIxDialogElement;
  onClose: TypedEvent<TReason>;
  onDismiss: TypedEvent<TReason>;
}

function getIxModal(element: Element) {
  return element.closest('ix-dialog');
}

export function closeModal<TClose = any>(
  element: Element,
  closeResult: TClose
) {
  const dialog = getIxModal(element);
  if (dialog) {
    dialog.closeModal(closeResult);
    return;
  }
}

export function dismissModal(element: Element, dismissResult?: any) {
  const dialog = getIxModal(element);
  if (dialog) {
    dialog.dismissModal(dismissResult);
    return;
  }
}

export async function showMessage<T>(config: MessageContent) {
  const onMessageAction = new TypedEvent<T>();
  const message = document.createElement('ix-modal-message');
  Object.assign(message, config);

  const dialog = document.createElement('ix-dialog');
  dialog.appendChild(message);
  const dialogRef = await getCoreDelegate().attachView(dialog);

  dialogRef.addEventListener('dialogClose', (event: CustomEvent<T>) => {
    onMessageAction.emit(event.detail);
    dialogRef.remove();
  });

  dialogRef.addEventListener('dialogDismiss', () => {
    dialogRef.remove();
  });

  return onMessageAction;
}

export async function showModal<T>(
  config: ModalConfig<T>
): Promise<ModalInstance<T>> {
  const delegate = resolveDelegate();
  let dialogRef: HTMLIxDialogElement;
  const onClose = new TypedEvent<T>();
  const onDismiss = new TypedEvent<T>();

  if (typeof config.content === 'string') {
    const dialog = document.createElement('ix-dialog');
    dialog.innerText = config.content;
    dialogRef = await getCoreDelegate().attachView(dialog);
  }

  if (
    config.content instanceof HTMLElement &&
    config.content.tagName !== 'IX-DIALOG'
  ) {
    const dialog = document.createElement('ix-dialog');
    dialog.appendChild(config.content);
    dialogRef = await getCoreDelegate().attachView(dialog);
  }

  if (!dialogRef) {
    dialogRef = await delegate.attachView<HTMLIxDialogElement>(config.content);
  }

  Object.assign(dialogRef, config);
  dialogRef.showModal();
  dialogRef.addEventListener('dialogClose', async ({ detail }: CustomEvent) => {
    onClose.emit(detail);
    await delegate.removeView(dialogRef);
  });

  dialogRef.addEventListener(
    'dialogDismiss',
    async ({ detail }: CustomEvent) => {
      onDismiss.emit(detail);
      await delegate.removeView(dialogRef);
    }
  );

  return {
    htmlElement: dialogRef,
    onClose,
    onDismiss,
  };
}
