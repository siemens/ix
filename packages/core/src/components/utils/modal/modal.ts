/*
 * SPDX-FileCopyrightText: 2023 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { IxModalSize } from '../../modal/modal';
import { getCoreDelegate, resolveDelegate } from '../delegate';
import { TypedEvent } from '../typed-event';

export function setA11yAttributes(element: HTMLElement, config: ModalConfig) {
  const ariaDescribedby = config.ariaDescribedby;
  const ariaLabelledby = config.ariaLabelledby;

  delete config['ariaDescribedby'];
  delete config['ariaLabelledby'];

  if (ariaDescribedby) {
    element.setAttribute('aria-describedby', ariaDescribedby);
  }

  if (ariaLabelledby) {
    element.setAttribute('aria-labelledby', ariaLabelledby);
  }
}

export interface ModalConfig<TReason = any, CONTENT = any> {
  animation?: boolean;
  ariaDescribedby?: string;
  ariaLabelledby?: string;
  backdrop?: boolean;
  closeOnBackdropClick?: boolean;
  beforeDismiss?: (reason?: TReason) => boolean | Promise<boolean>;
  centered?: boolean;
  container?: string | HTMLElement;
  content: CONTENT | string;
  keyboard?: boolean;
  size?: IxModalSize;
  title?: string;
}

export interface ModalInstance<TReason = any> {
  htmlElement: HTMLIxModalElement;
  onClose: TypedEvent<TReason>;
  onDismiss: TypedEvent<TReason>;
}

function getIxModal(element: Element) {
  return element.closest('ix-modal');
}

export function closeModal<TClose = any>(
  element: Element,
  closeResult: TClose
) {
  const dialog = getIxModal(element);
  if (dialog) {
    dialog.closeModal(closeResult);
    return;
  }
}

export function dismissModal(element: Element, dismissResult?: any) {
  const dialog = getIxModal(element);
  if (dialog) {
    dialog.dismissModal(dismissResult);
    return;
  }
}

export async function showModal<T>(
  config: ModalConfig<T>
): Promise<ModalInstance<T>> {
  const delegate = resolveDelegate();
  let dialogRef: HTMLIxModalElement;
  const onClose = new TypedEvent<T>();
  const onDismiss = new TypedEvent<T>();

  if (typeof config.content === 'string') {
    const dialog = document.createElement('ix-modal');
    dialog.innerText = config.content;
    dialogRef = await getCoreDelegate().attachView(dialog);
  }

  if (
    config.content instanceof HTMLElement &&
    config.content.tagName !== 'IX-MODAL'
  ) {
    const dialog = document.createElement('ix-modal');
    dialog.appendChild(config.content);
    dialogRef = await getCoreDelegate().attachView(dialog);
  }

  if (!dialogRef) {
    dialogRef = await delegate.attachView<HTMLIxModalElement>(config.content);
  }

  setA11yAttributes(dialogRef, config);
  Object.assign(dialogRef, config);

  dialogRef.setAttribute('ariaDescribedby', 'Test');

  dialogRef.showModal();
  dialogRef.addEventListener('dialogClose', async ({ detail }: CustomEvent) => {
    onClose.emit(detail);
    await delegate.removeView(dialogRef);
  });

  dialogRef.addEventListener(
    'dialogDismiss',
    async ({ detail }: CustomEvent) => {
      onDismiss.emit(detail);
      await delegate.removeView(dialogRef);
    }
  );

  return {
    htmlElement: dialogRef,
    onClose,
    onDismiss,
  };
}
