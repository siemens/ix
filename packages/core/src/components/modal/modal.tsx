/*
 * SPDX-FileCopyrightText: 2022 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  Host,
  Method,
  Prop,
} from '@stencil/core';
import anime from 'animejs';
import Animation from '../utils/animation';
import { NotificationColor } from '../utils/notification-color';

@Component({
  tag: 'ix-modal',
  styleUrl: 'modal.scss',
  scoped: true,
})
export class Modal {
  @Element() hostElement: HTMLIxModalElement;

  /**
   * Should the modal be animtated
   */
  @Prop() animation = true;

  /**
   *
   */
  @Prop() ariaDescribedBy: string;

  /**
   *
   */
  @Prop() ariaLabelledBy = 'modal-title';

  /**
   * Adds a dimming layer to the modal.
   * This should only be used when it it necessary to focus the user's attention to the dialog content (e.g. errors, warnings, complex tasks).
   */
  @Prop() backdrop: boolean | 'static' = true;

  /**
   * Backdrop class
   */
  @Prop() backdropClass: string;

  /**
   * BeforeDismiss callback
   */
  @Prop() beforeDismiss: (reason?: any) => boolean | Promise<boolean>;

  /**
   * Centered modal
   */
  @Prop() centered = false;

  /**
   * Content of modal
   */
  @Prop() content: HTMLElement | string;

  /**
   * ESC close modal dialog
   */
  @Prop() keyboard = true;

  /**
   * Optional icon displayed next to the title
   */
  @Prop() icon: string;

  /**
   * Color of the header {@see ix-icon}
   */
  @Prop() iconColor: NotificationColor = 'color-std-text';

  /**
   * Modal dialog class
   */
  @Prop() modalDialogClass: string;

  /**
   * Modal scollable
   */
  @Prop() scrollable = true;

  /**
   * Modal size
   */
  @Prop() size: 'sm' | 'lg' | 'xl' = 'sm';

  /**
   * Header title
   */
  @Prop() headerTitle: string;

  /**
   * Window class
   */
  @Prop() windowClass: string;

  /**
   * Modal closed
   */
  @Event() closed: EventEmitter;

  /**
   * Modal dismissed
   */
  @Event() dismissed: EventEmitter;

  private readonly onKeydown = this.handleKeydown.bind(this);

  get modal() {
    return this.hostElement.querySelector('.modal');
  }

  get modalDialog() {
    return this.modal.querySelector('.modal-dialog');
  }

  get modalContent() {
    return this.modalDialog.querySelector('.modal-content');
  }

  get modalBackdrop() {
    return this.hostElement.querySelector('.backdrop');
  }

  private slideDown(modal: any) {
    const duration = this.animation ? Animation.mediumTime : 0;

    anime({
      targets: modal,
      duration,
      opacity: [0, 1],
      translateY: ['-100%', 0],
      easing: 'easeOutSine',
    });
  }

  private slideUp(modal: any, completeCallback?: any) {
    const duration = this.animation ? Animation.mediumTime : 0;

    anime({
      targets: modal,
      duration,
      opacity: [1, 0],
      translateY: [0, '-100%'],
      easing: 'easeInSine',
      complete: () => {
        if (completeCallback) {
          completeCallback();
        }
      },
    });
  }

  private onBackdropClick(event: Event) {
    const target = event.target as Element;
    if (target.classList.contains('backdrop')) {
      this.dismiss(event);
    }
  }

  componentDidLoad() {
    if (this.backdrop === 'static') {
      this.modalBackdrop.addEventListener('click', (event) =>
        this.onBackdropClick(event)
      );
    }

    if (this.backdropClass) {
      this.modalBackdrop.classList.add(this.backdropClass);
    }

    if (this.modalDialogClass) {
      this.modalDialog.classList.add(this.modalDialogClass);
    }

    if (this.windowClass) {
      this.modal.classList.add(this.windowClass);
    }

    if (this.keyboard) {
      window.addEventListener('keydown', this.onKeydown);
    }

    this.slideDown(this.modalContent);
  }

  private handleKeydown(ev: KeyboardEvent) {
    if (ev.key === 'Escape') {
      this.dismiss(ev.key);
    }
  }

  disconnectedCallback() {
    window.removeEventListener('keydown', this.onKeydown);
  }

  /**
   * Dismiss modal instance
   * @param reason
   */
  @Method()
  async dismiss<T = any>(reason?: T) {
    if (this.beforeDismiss) {
      const result = await this.beforeDismiss(reason);
      if (result !== false) {
        this.slideUp(this.modalContent, () => this.dismissed.emit(reason));
      }
    } else {
      this.slideUp(this.modalContent, () => this.dismissed.emit(reason));
    }
  }

  /**
   * Close modal
   * @param result
   */
  @Method()
  async close<T = any>(result: T) {
    this.slideUp(this.modalContent, () => this.closed.emit(result));
  }

  render() {
    return (
      <Host>
        <div
          class={{
            animation: this.animation,
            modal: true,
            backdrop: this.backdrop === 'static' || this.backdrop,
            'align-items-center': this.centered,
            scrollable: this.scrollable,
          }}
          aria-describedby={this.ariaDescribedBy}
          aria-labelledby={this.ariaLabelledBy}
        >
          <div
            class={{
              'modal-dialog': true,
              'modal-sm': this.size === 'sm',
              'modal-lg': this.size === 'lg',
              'modal-xl': this.size === 'xl',
            }}
          >
            <div class="modal-content">
              {this.icon === undefined || this.icon === '' ? (
                ''
              ) : (
                <div class="state-icon-container">
                  <ix-icon
                    name={this.icon}
                    size="32"
                    color={this.iconColor}
                  ></ix-icon>
                </div>
              )}
              <div class="slot-container">
                <slot></slot>
              </div>
            </div>
          </div>
        </div>
      </Host>
    );
  }
}
