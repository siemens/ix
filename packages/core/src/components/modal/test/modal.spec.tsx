/*
 * SPDX-FileCopyrightText: 2022 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { newSpecPage } from '@stencil/core/testing';
import { Modal } from '../modal';

describe('modal', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [Modal],
      html: '<ix-modal></ix-modal>',
    });
    expect(page.root).toMatchSnapshot();
  });
});
