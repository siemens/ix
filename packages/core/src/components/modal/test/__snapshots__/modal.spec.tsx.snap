// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal renders 1`] = `
<ix-modal id="ix-modal-instance-1">
  <!---->
  <div aria-labelledby="modal-title" class="animation backdrop modal scrollable">
    <div class="modal-dialog modal-sm">
      <div class="modal-content">
        <div class="slot-container"></div>
      </div>
    </div>
  </div>
</ix-modal>
`;
